Source/WebCore/html/parser/HTMLConstructionSite.cpp

```diff
@@ -72,30 +72,32 @@ static inline void setAttributes(Element& element, AtomHTMLToken& token, ParserC
 
 static bool hasImpliedEndTag(const HTMLStackItem& item)
 {
-    return item.hasTagName(ddTag)
-        || item.hasTagName(dtTag)
-        || item.hasTagName(liTag)
-        || is<HTMLOptionElement>(item.node())
-        || is<HTMLOptGroupElement>(item.node())
-        || item.hasTagName(pTag)
-        || item.hasTagName(rbTag)
-        || item.hasTagName(rpTag)
-        || item.hasTagName(rtTag)
-        || item.hasTagName(rtcTag);
+    return item.elementName() == ElementName::HTML_dd
```

Comment:
Can we use `ElementName::HTML::dd` instead?

Reply:
We could do that with if the name's slightly different, for example:

```c++
enum class ElementName : uint16_t {
....
    HTML_dd,
    SVG_path,
    ...
};

namespace ElementNames {
namespace HTML {
constexpr ElementName dd = ElementName::HTML_dd;
...
}
namespace SVG {
constexpr ElementName path = ElementName::HTML_path;
}
}
```

then you could write either `ElementName::HTML_dd` or `ElementNames::HTML::dd`. Not sure if that's confusing.

But I'm sympathetic to the goal of reducing visual noise, so I'm happy to entertain suggestions.

Reply:
Another option:

```c++
enum class ElementName : uint16_t {
    ...
    HTML_dd,
    SVG_path,
    ...
};

namespace ElementNameHTML {
constexpr ElementName dd = ElementName::HTML_dd;
...
}
namespace ElementNameSVG {
constexpr ElementName path = ElementName::SVG_path;
...
}
}
```

then we can write `ElementNameHTML::dd` in place of `ElementName::HTML_dd`. @rniwa do you like either of these, or have any other suggestion?

Reply:
I've done something like this, and I think it reads well. I've left it as a separate commit on top of the main commit, in case I change my mind.